File: backend/itinerary-service/src/main/java/com/tripify/itinerary_service/dto/AddToCartRequestDTO.java
package com.tripify.itinerary_service.dto;

import java.time.LocalDate;

// Rispecchia esattamente AddToCartRequestDTO di booking-service (POST /api/v1/cart/add):
// è il corpo che ItineraryService.bookAllItems inoltra per ogni componente della lista.
public record AddToCartRequestDTO(
        Long catalogItemId,
        Integer quantity,
        Long roomTypeId,
        Long fareClassId,
        LocalDate checkIn,
        LocalDate checkOut
) {}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/AddToCartRequestDTO.kt
package com.tripify.tripify_android.data.model

data class AddToCartRequestDTO(
    val catalogItemId: Long,
    val quantity: Int,
    val roomTypeId: Long? = null,
    val fareClassId: Long? = null,
    val checkIn: String? = null,
    val checkOut: String? = null
)
